### PR TITLE
New Spanner query results to Elasticsearch (batch) template

### LINKED ISSUE #456: [Feature Request]: Add batch template to support loading Spanner DB to Elasticsearch
### Related Template(s)

googlecloud-to-elasticsearch

### What feature(s) are you requesting?

I created a streaming template for loading Spanner change stream data into Elasticsearch (PR #429 )

This a request for another template to batch load data from an existing Spanner DB to Elasticsearch

### PR DESCRIPTION
**Pull request by ggprod**
Resolves #456

### CODE REVIEW

File: v2/googlecloud-to-elasticsearch/src/main/java/com/google/cloud/teleport/v2/elasticsearch/options/SpannerToElasticsearchOptions.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  @Default.Enum("HIGH")
+  RpcPriority getSpannerRpcPriority();
+
+  void setSpannerRpcPriority(RpcPriority spannerRpcPriority);
+
+  @Description("Required. Query used to read from Spanner table.")
+  @Required
+  String getSqlQuery();
+
+  void setSqlQuery(String sqlQuery);
```
Comment (bvolpato) [inline v2/googlecloud-to-elasticsearch/src/main/java/com/google/cloud/teleport/v2/elasticsearch/options/SpannerToElasticsearchOptions.java:52]: Could those fields possibly come out to a reusable `SpannerReadOptions`?

As of now, `SpannerToElasticsearchOptions` and `SpannerToBigQueryOptions` only differ by a single field (`bigQuerySchemaPath`)

File: v2/googlecloud-to-elasticsearch/src/test/java/com/google/cloud/teleport/v2/elasticsearch/templates/SpannerToElasticsearchTest.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  private static final long b = 1001;
+  private static final Struct struct = Struct.newBuilder().set("a").to(a).set("b").to(b).build();
+  private static final List<Struct> rows = ImmutableList.of(struct);
+  private static final String jsonifiedTableRow = "{\"a\":\"testing\",\"b\":1001}";
+  @Rule public final transient TestPipeline pipeline = TestPipeline.create();
+  @Rule public ExpectedException exceptionRule = ExpectedException.none();
+
+  /** Test the {@link SpannerToElasticsearch} pipeline end-to-end. */
+  @Test
+  public void testSpannerToElasticsearchE2E() {
```
Comment (bvolpato) [inline v2/googlecloud-to-elasticsearch/src/test/java/com/google/cloud/teleport/v2/elasticsearch/templates/SpannerToElasticsearchTest.java:46]: Like on the Spanner case, it's ok not to be E2E.

File: v2/googlecloud-to-elasticsearch/src/main/java/com/google/cloud/teleport/v2/elasticsearch/options/SpannerToElasticsearchOptions.java (hunk 1 above)
Comment (ggprod) [reply v2/googlecloud-to-elasticsearch/src/main/java/com/google/cloud/teleport/v2/elasticsearch/options/SpannerToElasticsearchOptions.java:52]: yes, I will make that change

File: v2/googlecloud-to-elasticsearch/src/test/java/com/google/cloud/teleport/v2/elasticsearch/templates/SpannerToElasticsearchTest.java (hunk 2 above)
Comment (ggprod) [reply v2/googlecloud-to-elasticsearch/src/test/java/com/google/cloud/teleport/v2/elasticsearch/templates/SpannerToElasticsearchTest.java:46]: yes, I will rename it appropriately as it is not E2E

File: v2/googlecloud-to-elasticsearch/src/test/java/com/google/cloud/teleport/v2/elasticsearch/templates/SpannerToElasticsearchTest.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+/** Test cases for {@link SpannerToElasticsearch}. */
+public class SpannerToElasticsearchTest {
+  private static final String a = "testing";
+  private static final long b = 1001;
+  private static final Struct struct = Struct.newBuilder().set("a").to(a).set("b").to(b).build();
+  private static final List<Struct> rows = ImmutableList.of(struct);
+  private static final String jsonifiedTableRow = "{\"a\":\"testing\",\"b\":1001}";
```
[the remark this answers is not in the document]
Comment (ggprod) [reply v2/googlecloud-to-elasticsearch/src/test/java/com/google/cloud/teleport/v2/elasticsearch/templates/SpannerToElasticsearchTest.java:40]: yes, will add more complex types

**Comment by bvolpato** [comment]
Hi @ggprod -- thanks for your contribution here. 

Unfortunately we ran behind on reviews / merge, and the structure of the project changed a bit. Is there still interest in pursuing this PR?

We can work together on a plan to resolve conflicts + review the code.

**Pull request closed without merging**